### ISSUE TITLE
Problems with resource spec and taggable resources

### ISSUE THREAD

**Issue by seporaitis** [aws-cloudformation/cfn-python-lint#1068]
*cfn-lint version: (`cfn-lint --version`)* 0.22.4

*Description of issue.*

We have a custom linter rule to enforce specific tags on all taggable resources, similar to this:

```python
from cfnlint import CloudFormationLintRule
from cfnlint import RuleMatch
from cfnlint.helpers import RESOURCE_SPECS


class AppStageTagsIncluded(CloudFormationLintRule):
    id = 'E9001'
    shortdesc = 'Checks for App and optional Stage tags on resources'
    description = shortdesc
    tags = ['resources', 'tags']

    def match(self, cfn):
        """Check Tags for required keys"""
        resource_types_with_tags = self.get_resource_types_with_tags(cfn.regions[0])
        resources = cfn.get_resources()

        tags_to_check = ['App']
        if 'Stage' in cfn.get_parameters():
            tags_to_check.append('Stage')

        matches = []
        for name, resource in resources.items():
            resource_type = resource.get('Type', '')
            if resource_type not in resource_types_with_tags:
                continue

            resource_properties = resource.get('Properties', {})
            tags = resource_properties.get('Tags', [])
            tags_dict = {i.get('Key'): i.get('Value') for i in tags}

            for tag in tags_to_check:
                if tag not in tags_dict:
                    matches.append(RuleMatch(
                        ['Resources', name, 'Properties', 'Tags'],
                        f'Missing tag "{tag}" for Resources/{name}',
                    ))

        return matches

    def get_resource_types_with_tags(self, region):
        all_resource_types = RESOURCE_SPECS[region]['ResourceTypes']

        resource_types = []
        for resource_type, resource in all_resource_types.items():
            properties = resource.get('Properties')
            if properties and 'Tags' in properties:
                resource_types.append(resource_type)
        return resource_types
```

Recently AWS IAM must have enabled tagging on `AWS::IAM::User` resources, because `RESOURCE_SPECS` identifies it as having `Tags` in `Properties`. However, in reality trying to deploy a template like this:

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  UserTest:
    Type: AWS::IAM::User
    Properties:
        UserName: test-user
        Tags:
        - Key: App
           Value: Users
        - Key: Stage
           Value: Test
```

CloudFormation returns an error trying to update `UserTest`: `Encountered unsupported property Tags`.

I verified that the `data/CloudSpecs/us-east-1.json` has `Tags` listed as possible property: 

```json
    "AWS::IAM::User": {
      "Attributes": {
        "Arn": {
          "PrimitiveType": "String"
        }
      },
      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iam-user.html",
      "Properties": {
        # ...
        "Tags": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-user.html#cfn-iam-user-tags",
          "DuplicatesAllowed": true,
          "ItemType": "Tag",
          "Required": false,
          "Type": "List",
          "UpdateType": "Mutable"
        },
        # ...
      }
    },
```

Since this is a custom rule, I wouldn't normally create an issue in this project, however our "source of truth" is the CloudFormation specification that `cfn-lint` uses. Hence, I would like advice which of the following could be happening:

* is this a problem on our end/we do something incorrectly;
* is the CloudFormation specification in the most recent version inaccurate; 
* is CloudFormation service not conforming with correct specification?

Thank you.

**Comment by kddejong** [comment]
The IAM resources have been an issue as of late with the tags.  What region are you deploying into?  I had thought that this was fully available in us-east-1, us-west-2 but not sure how widely deployed it is.

**Comment by kddejong** [comment]
Here is my current view of what is probably supported.

- us-east-1 - Supported
- us-east-2 - Not Supported  
- us-west-1 - Not Supported  
- us-west-2 - Supported
- ca-central-1 - Not Supported  
- eu-central-1  - Supported
- eu-west-1 - Supported
- eu-west-2  - Not Supported  
- ap-northeast-1 - Supported
- ap-northeast-2 - Supported
- ap-southeast-1 - Supported
- ap-southeast-2 - Supported
- ap-south-1  - Not Supported  
- sa-east-1 - Supported
- eu-west-3 - Not Supported  
- eu-north-1 - Not Supported

**Comment by kddejong** [comment]
@seporaitis if you could validate the region you are using I would appreciate it.  I'm hoping it falls into one of my Not Supported regions.  If it does match I can clean up the tagging pretty quick and get it to better match reality.

**Comment by seporaitis** [comment]
@kddejong apologies for delayed response on my part. The region in question is `us-east-1` which should be supported.

**Comment by seporaitis** [comment]
So the above pasted snipped from the spec included with `cfn-lint` has `Tags`, but when checking `us-east-1` in the spec of the official [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) - it seems to be missing. Is it possible that one of them was not updated or was reverted?

**Comment by kddejong** [comment]
Oh yea, for that we patch that in.  https://github.com/aws-cloudformation/cfn-python-lint/blob/v0.22.4/src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json#L355-L366

This is because from everything we have seen it is working in us-east-1.  Problem is that patch goes to all regions, which isn't supported currently, so I think I need to move this patch into the appropriate regions until its fully supported across all regions.

There has been a lot of weird things with this one in particular were it was in, out, works in some regions, etc. that has made this one in particular an issue.

**Comment by kddejong** [comment]
Oops and I apologize I tested Role and not User so going to try a user test too

**Comment by kddejong** [comment]
So, me testing the Role and not the User really messed me up.  User Tags seems completely unsupported across the board.  I'll get that cleaned up and sorry for my confusion.

**Comment by seporaitis** [comment]
Things happen :) So thank you for reacting to this so quickly, @kddejong !

**Comment by kddejong** [comment]
@seporaitis released with v0.23.0.
